pkg/chunk/chunk_test.go

```diff
@@ -132,13 +135,30 @@ func TestChunkCodec(t *testing.T) {
 const fixedTimestamp = model.Time(1557654321000)
 
 func TestChunkDecodeBackwardsCompatibility(t *testing.T) {
+	// lets build a new chunk same as what was built using code at commit b1777a50ab19
+	c, _ := encoding.NewForEncoding(encoding.Bigchunk)
+	nc, err := c.Add(model.SamplePair{Timestamp: fixedTimestamp, Value: 0})
+	require.NoError(t, err)
+	require.Equal(t, nil, nc, "returned chunk should be nil")
+
+	chunk := NewChunk(
+		userID,
+		client.Fingerprint(labelsForDummyChunks),
+		labelsForDummyChunks,
+		c,
+		fixedTimestamp.Add(-time.Hour),
+		fixedTimestamp,
+	)
+	// Force checksum calculation.
+	require.NoError(t, chunk.Encode())
```

Comment:
Thanks for changing this based on my previous comment. I was wondering why of the previous comment but I'm not against changing the fixture if you build the fixtures based on the commit `b1777a50ab19`. I think the initial version of the change was clearer instead of copy-pasting code from `b1777a50ab19`.

Reply:
I didn't copy any code from `b1777a50ab19`. This code just includes relevant parts from `dummyChunkForEncoding`. I think this change makes it easy for us to freely change that code in future and avoid breaking backwards compatibility accidentally. Someone could err in building a chunk from that old commit and tests might still work but we realize something is wrong only in future. Please let me know what do you think.

Reply:
You're right. LGTM.